R/matrix.R

```diff
@@ -2385,9 +2385,27 @@ setAs("IterableMatrix", "matrix", function(from) {
   rlang::inform(c(
       "Warning: Converting to a dense matrix may use excessive memory"
     ), .frequency = "regularly", .frequency_id = "matrix_dense_conversion")
-  as(from, "dgCMatrix") %>% as.matrix()  
+  # `mat` will always be numeric mode
+  mat <- as.matrix(as(from, "dgCMatrix"))
+  # to keep the original mode, we transform it when necessary
+  if (matrix_type(from) == "uint32_t") {
+      mat <- matrix_to_integer(mat)
+  }
+  mat
 })
 
+matrix_to_integer <- function(matrix) { # a numeric matrix
+    if (is.integer(matrix)) return(matrix) # styler: off
+    if (all(matrix <= .Machine$integer.max)) {
+        storage.mode(matrix) <- "integer"
+    } else {
+        cli::cli_warn(
```

Comment:
Could we change this to a call to `rlang::warn`? I don't think `cli` is in the dependency list right now

Reply:
It's well to do that, I missed it

Reply:
My working computer is not at hand, I'll do it tonight.

Reply:
No rush, I just wanted to get you back comments in a semi-timely manner :)